app/services/katello/repository_type_manager.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class RepositoryTypeManager
```

Comment:
I like your Bastion  plugins example actually and prefer it. But I was recommended to use the foreman plugin model.